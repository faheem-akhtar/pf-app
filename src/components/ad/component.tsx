import { FunctionComponent } from 'react';

import styles from './ad.module.scss';

export const AdComponent: FunctionComponent<{ id: string }> = ({ id }) => <div data-ad className={styles.ad} id={id} />;
