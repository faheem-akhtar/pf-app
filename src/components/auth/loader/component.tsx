import { domClassMerge } from 'helpers/dom/class-merge';

import styles from './auth-loader.module.scss';

export const AuthLoaderComponent = ({
  isEnabled,
  isCentered,
}: {
  isEnabled: boolean;
  isCentered: boolean;
}): JSX.Element | null =>
  isEnabled ? (
    <div className={styles.loader1}>
      <div
        className={domClassMerge(styles['loader1__spinner'], isCentered ? styles['loader1__spinner--centered'] : '')}
      >
        <div className={domClassMerge(styles['loader1__bounce'], styles['loader1__bounce--1'])} />
        <div className={domClassMerge(styles['loader1__bounce'], styles['loader1__bounce--2'])} />
        <div className={domClassMerge(styles['loader1__bounce'], styles['loader1__bounce--3'])} />
      </div>
    </div>
  ) : null;
