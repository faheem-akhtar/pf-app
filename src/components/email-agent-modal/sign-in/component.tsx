import { useContext, useRef } from 'react';

import { UserContext } from 'context/user/context';

import { EmailAgentModalSignInComponentPropsInterface } from './component-props.interface';
import { EmailAgentModalSignInTemplate } from './template';

export const EmailAgentModalSignInComponent = (
  props: EmailAgentModalSignInComponentPropsInterface
): JSX.Element | null => {
  const openAuthRef = useRef<() => void>(() => null);
  const closeAuthRef = useRef<() => void>(() => null);

  const user = useContext(UserContext);

  if (user) return null;

  return <EmailAgentModalSignInTemplate {...props} openAuthRef={openAuthRef} closeAuthRef={closeAuthRef} />;
};
