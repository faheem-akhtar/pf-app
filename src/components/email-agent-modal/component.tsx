import { FunctionComponent, useContext, useRef, useState } from 'react';

import { apiEmailAgentFetcher } from 'api/email-agent/fetcher';
import { AuthModalComponent } from 'components/auth/modal/component';
import { formMakeValidator } from 'components/form/make-validator';
import { IconThickSmallCloseTemplate } from 'components/icon/thick/small-close-template';
import { ModalComponent } from 'components/modal/component';
import { propertySerpObfuscatedGetId } from 'components/property/serp/obfuscated/get/id';
import { propertySerpObfuscatedGetName } from 'components/property/serp/obfuscated/get/name';
import { propertySerpObfuscatedGetReference } from 'components/property/serp/obfuscated/get/reference';
import { configEmailAgentEmailAlertCheckedByDefault } from 'config/email-agent/email-alert-checked-by-default';
import { configEmailAgentMustAcceptTerms } from 'config/email-agent/must-accept-terms';
import { configEmailAgentReceiveAdsEnabled } from 'config/email-agent/receive-ads-enabled';
import { UserContext } from 'context/user/context';
import { functionNoop } from 'helpers/function/noop';
import { useTranslation } from 'helpers/translation/hook';
import { validationEmail } from 'helpers/validation/email';
import { validationPhone } from 'helpers/validation/phone';
import { validationRequired } from 'helpers/validation/required';
import { GoogleRecaptchaService } from 'services/google/recaptcha.service';
import { EmailAgentAttributesInterface } from 'types/email-agent/attributes.interface';

import { EmailAgentModalComponentPropsInterface } from './component-props.interface';
import styles from './email-agent-modal.module.scss';
import { EmailAgentModalFormComponent } from './form/component';
import { EmailAgentModalFormComponentPropsInterface } from './form/component-props.interface';
import { EmailAgentModalFormErrorMessageTemplate } from './form/error-message-template';
import { FormFieldsEnum } from './form/fields.enum';
import { FormFieldsValueType } from './form/fields-value.type';
import { EmailAgentModalFormSuccessTemplate } from './form/success-template';
import { EmailAgentModalStatusEnum } from './status.enum';

const captchaService = GoogleRecaptchaService();

export const EmailAgentModalComponent: FunctionComponent<EmailAgentModalComponentPropsInterface> = ({
  openRef,
  property,
}) => {
  const { t } = useTranslation();
  const user = useContext(UserContext);

  const openAuthRef = useRef<() => void>(functionNoop);
  const closeAuthRef = useRef<() => void>(functionNoop);
  const closeRef = useRef<() => void>(functionNoop);

  const getInitialFieldsValue = (): FormFieldsValueType => ({
    name: user ? `${user.first_name} ${user.last_name}` : '',
    email: user ? user.email : '',
    phone: '',
    message: t('agent-modal/default-email-message').replace('#{ref}', propertySerpObfuscatedGetReference(property)),
    emailAlert: configEmailAgentEmailAlertCheckedByDefault,
    acceptTerms: false,
    receiveAdvertising: false,
  });
  const [fieldsValue, setFieldsValue] = useState<FormFieldsValueType>(getInitialFieldsValue());
  const [errors, setErrors] = useState<Partial<Record<FormFieldsEnum, string>>>({});
  const [error, setError] = useState('');
  const [isLoading, setIsLoading] = useState(false);
  const [validators] = useState(() => {
    const state: Record<string, Array<(value: string) => string>> = {
      [FormFieldsEnum.name]: [validationRequired(t('validation/empty-name'))],
      [FormFieldsEnum.email]: [
        validationRequired(t('validation/empty-email')),
        validationEmail(t('validation/not-valid-email')),
      ],
      [FormFieldsEnum.phone]: [
        validationRequired(t('validation/empty-phone-number')),
        validationPhone(t('validation/not-valid-phone-number')),
      ],
      [FormFieldsEnum.message]: [validationRequired(t('validation/empty-message'))],
    };

    if (configEmailAgentMustAcceptTerms) {
      state[FormFieldsEnum.acceptTerms] = [validationRequired(t('agent-modal/accept-conditions-error-message'))];
    }
    if (configEmailAgentReceiveAdsEnabled) {
      state[FormFieldsEnum.receiveAdvertising] = [validationRequired(t('agent-modal/accept-conditions-error-message'))];
    }
    return state;
  });
  const [status, setStatus] = useState<EmailAgentModalStatusEnum>(EmailAgentModalStatusEnum.opened);
  const validate = formMakeValidator(errors, setErrors, validators);

  const onOpen = (): void => {
    setFieldsValue(getInitialFieldsValue());
    setErrors({});
    setStatus(EmailAgentModalStatusEnum.opened);
  };

  const closeModal = (): void => {
    closeRef.current();
  };

  const onSubmit: EmailAgentModalFormComponentPropsInterface['onSubmit'] = async (e) => {
    e.preventDefault();
    setIsLoading(true);

    const formFields: Record<string, string | boolean> = {
      [FormFieldsEnum.name]: fieldsValue.name,
      [FormFieldsEnum.email]: fieldsValue.email,
      [FormFieldsEnum.phone]: fieldsValue.phone,
      [FormFieldsEnum.message]: fieldsValue.message,
    };

    if (configEmailAgentMustAcceptTerms) {
      formFields[FormFieldsEnum.acceptTerms] = fieldsValue.acceptTerms;
    }
    if (configEmailAgentReceiveAdsEnabled) {
      formFields[FormFieldsEnum.receiveAdvertising] = fieldsValue.receiveAdvertising;
    }

    const hasErrors = validate(formFields);

    if (hasErrors) {
      setIsLoading(false);
      return;
    }

    setStatus(EmailAgentModalStatusEnum.submitting);
    setError('');

    try {
      const captcha_token = await captchaService.execute();

      await apiEmailAgentFetcher({
        propertyId: parseInt(propertySerpObfuscatedGetId(property), 10),
        ...formFields,
        emailAlert: fieldsValue.emailAlert,
        [FormFieldsEnum.captchaToken]: captcha_token,
        autoRegister: !!user,
      } as EmailAgentAttributesInterface);

      setStatus(EmailAgentModalStatusEnum.submitted);
    } catch {
      setError(t('something-wrong-try-again'));
      captchaService.reset();
    } finally {
      setIsLoading(false);
    }
  };

  const body =
    status === EmailAgentModalStatusEnum.submitted ? (
      <EmailAgentModalFormSuccessTemplate closeModal={closeModal} t={t} openAuthRef={openAuthRef} />
    ) : (
      <>
        <h2 className={styles.name}>{propertySerpObfuscatedGetName(property)}</h2>
        <EmailAgentModalFormComponent
          fieldsValue={fieldsValue}
          setFieldsValue={setFieldsValue}
          onSubmit={onSubmit}
          t={t}
          errors={errors}
          loading={isLoading}
        />
      </>
    );

  return (
    <>
      <ModalComponent
        onOpen={onOpen}
        openRef={openRef}
        closeRef={closeRef}
        overlay
        onOverlayClick={(): void => closeRef.current()}
      >
        <div
          data-testid='email-agent-modal-content'
          className={styles.container}
          onClick={(e): void => {
            e.stopPropagation();
          }}
        >
          <header className={styles.header}>
            <h1 className={styles.title}>{t('agent-modal/email-form-title')}</h1>
            <div className={styles.closeButton} onClick={closeModal}>
              <IconThickSmallCloseTemplate class={styles.closeIcon} clipped />
            </div>
          </header>

          {error && <EmailAgentModalFormErrorMessageTemplate error={error} />}
          {body}
        </div>
      </ModalComponent>
      <ModalComponent openRef={openAuthRef} closeRef={closeAuthRef} overlay>
        <AuthModalComponent
          close={(): void => {
            closeAuthRef.current();
          }}
        />
      </ModalComponent>
    </>
  );
};
