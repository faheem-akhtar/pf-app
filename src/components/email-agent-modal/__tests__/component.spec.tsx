/**
 * @jest-environment jsdom
 */

import { act, render, RenderResult, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { mockModalEnv } from 'mocks/modal-env/mock';
import { mockReactUseSwr } from 'mocks/react/use-swr.mock';
import { mockWindowFetch } from 'mocks/window/fetch.mock';
import { propertyStub } from 'stubs/property/stub';
import { userModelStub } from 'stubs/user/model.stub';

import { propertySerpObfuscatedGetId } from 'components/property/serp/obfuscated/get/id';
import { PropertySerpObfuscatedType } from 'components/property/serp/obfuscated/type';
import { UserContext } from 'context/user/context';
import * as GoogleRecaptchaServiceModule from 'services/google/recaptcha.service';
import { StatsService } from 'services/stats/service';

import { EmailAgentModalComponent } from '../component';
import { EmailAgentModalComponentPropsInterface } from '../component-props.interface';

describe('EmailAgentModalComponent', () => {
  const openRef = { current: jest.fn() };
  const closeRef = { current: jest.fn() };
  const openAuthRef = { current: jest.fn() };
  const closeAuthRef = { current: jest.fn() };

  const property: PropertySerpObfuscatedType = propertyStub();
  let props: EmailAgentModalComponentPropsInterface;

  beforeAll(() => {
    props = {
      property,
      openRef,
    };
  });

  beforeEach(() => {
    (StatsService().propertyLeadSend as jest.Mock).mockReset();

    mockModalEnv();
    mockReactUseSwr('en-countries-GET-{"sort":"priority"}', {
      ok: true,
      data: [
        { code: 'ae', name: 'United Arab Emirates', phoneCode: '+971' },
        { code: 'tr', name: 'Turkey', phoneCode: '+90' },
      ],
    });
  });

  it('should take initial values when modal is opened', () => {
    render(
      <UserContext.Provider value={userModelStub()}>
        <EmailAgentModalComponent {...props} />
      </UserContext.Provider>
    );
    act(openRef.current);

    expect(screen.getByTestId('email-agent-form')).toHaveFormValues({
      name: 'FirstName LastName',
      email: 'test@propertyfinder.ae',
      phone: '',
      message: 'agent-modal/default-email-message',
      emailAlert: true,
    });
  });

  it('should not appear sign in pop-up', async () => {
    render(
      <UserContext.Provider value={userModelStub()}>
        <EmailAgentModalComponent {...props} />
      </UserContext.Provider>
    );
    act(openRef.current);

    userEvent.type(screen.getByPlaceholderText('phone'), '123456');

    GoogleRecaptchaServiceModule.GoogleRecaptchaService().execute = (): Promise<string> => Promise.resolve('token');

    userEvent.click(screen.getByRole('checkbox', { name: 'agent-modal/email-alert-message' }));
    userEvent.click(screen.getByRole('button', { name: 'agent-modal/cta-send-message' }));

    const fetchMock = mockWindowFetch();
    await waitFor(() =>
      expect(fetchMock).toHaveBeenCalledWith(
        'default-origin/en/api/pwa/property/email-agent',
        expect.objectContaining({
          body: JSON.stringify({
            attributes: {
              propertyId: parseInt(propertySerpObfuscatedGetId(props.property), 10),
              name: 'FirstName LastName',
              email: 'test@propertyfinder.ae',
              phone: '+971123456',
              message: 'agent-modal/default-email-message',
              emailAlert: false,
              captcha_token: 'token',
              autoRegister: true,
            },
          }),
          headers: {
            locale: 'en',
          },
          method: 'POST',
        })
      )
    );

    expect(screen.queryByText('agent-modal/sign-in-title')).not.toBeInTheDocument();
  });

  it('should the modal disappear when overlay is clicked', async () => {
    render(
      <UserContext.Provider value={userModelStub()}>
        <EmailAgentModalComponent {...props} />
      </UserContext.Provider>
    );
    act(openRef.current);

    const modalOverlay = screen.getByRole('document');
    userEvent.click(modalOverlay);

    act(closeRef.current);
    await waitFor(() => expect(screen.queryByText('agent-modal/email-form-title')).not.toBeInTheDocument());
  });

  describe('when the user is not exist', () => {
    let renderResult: RenderResult;

    beforeEach(() => {
      renderResult = render(
        <UserContext.Provider value={null}>
          <EmailAgentModalComponent {...props} />
        </UserContext.Provider>
      );
      act(openRef.current);
    });

    it('should render without throwing any errors', () => {
      expect(screen.getByTestId('email-agent-modal-content')).toMatchSnapshot();
    });

    it('should display property data properly', () => {
      expect(screen.getByRole('heading', { level: 2 })).toHaveTextContent('Title from agent');
      expect(screen.getByLabelText('message')).toHaveValue('agent-modal/default-email-message');
    });

    it('should take initial values when modal is opened', () => {
      expect(screen.getByTestId('email-agent-form')).toHaveFormValues({
        name: '',
        email: '',
        phone: '',
        message: 'agent-modal/default-email-message',
        emailAlert: true,
      });
      expect(screen.queryByLabelText('agent-modal/receive-advertising-message')).not.toBeInTheDocument();
      expect(screen.queryByLabelText('agent-modal/accept-terms-message-prefix')).not.toBeInTheDocument();

      const errorTexts = screen.queryAllByLabelText(/^validation(?:_?[a-z]+)*/);
      expect(errorTexts).toHaveLength(0);
    });

    describe('form submission', () => {
      it('should form successfully submitted', async () => {
        userEvent.type(screen.getByLabelText('name'), 'Name');
        userEvent.type(screen.getByLabelText('email'), 'email@example.com');
        userEvent.type(screen.getByPlaceholderText('phone'), '123456');

        GoogleRecaptchaServiceModule.GoogleRecaptchaService().execute = (): Promise<string> => Promise.resolve('token');

        userEvent.click(screen.getByRole('button', { name: 'agent-modal/cta-send-message' }));

        const fetchMock = mockWindowFetch();
        await waitFor(() =>
          expect(fetchMock).toHaveBeenCalledWith(
            'default-origin/en/api/pwa/property/email-agent',
            expect.objectContaining({
              body: JSON.stringify({
                attributes: {
                  propertyId: parseInt(propertySerpObfuscatedGetId(props.property), 10),
                  name: 'Name',
                  email: 'email@example.com',
                  phone: '+971123456',
                  message: 'agent-modal/default-email-message',
                  emailAlert: true,
                  captcha_token: 'token',
                  autoRegister: false,
                },
              }),
              headers: {
                locale: 'en',
              },
              method: 'POST',
            })
          )
        );

        expect(StatsService().propertyLeadSend).toHaveBeenCalledTimes(1);
        expect(StatsService().propertyLeadSend).toHaveBeenCalledWith(198023, {
          lead: {
            medium: 'email',
            cta: 'button',
            email: {
              name: 'Name',
              email: 'email@example.com',
              phone: '+971123456',
              message: 'agent-modal/default-email-message',
              emailAlert: true,
            },
          },
        });

        expect(screen.queryByTestId('loader-template')).not.toBeInTheDocument();
        expect(screen.queryByRole('heading', { level: 2 })).not.toBeInTheDocument();
        expect(screen.getByText('thank-you')).toBeInTheDocument();

        expect(screen.getByTestId('email-agent-modal-content')).toMatchSnapshot();
        expect(screen.getByText('agent-modal/sign-in-title')).toBeInTheDocument();

        userEvent.click(screen.getByRole('button', { name: /sign-in/i }));
        act(closeRef.current);
        act(openAuthRef.current);

        await waitFor(() => expect(screen.getByRole('heading', { level: 1, name: 'sign-in' })).toBeInTheDocument());
        expect(screen.queryByTestId('email-agent-modal-content')).not.toBeInTheDocument();

        userEvent.click(screen.getByRole('button', { name: /cross/i }));
        act(closeAuthRef.current);
      });

      it('should display API error message', async () => {
        userEvent.type(screen.getByLabelText('name'), 'Name');
        userEvent.type(screen.getByLabelText('email'), 'email@example.com');
        userEvent.type(screen.getByPlaceholderText('phone'), '123456');

        GoogleRecaptchaServiceModule.GoogleRecaptchaService().execute = (): Promise<string> => Promise.resolve('token');

        userEvent.click(screen.getByRole('button', { name: 'agent-modal/cta-send-message' }));

        await waitFor(() =>
          renderResult.rerender(
            <UserContext.Provider value={null}>
              <EmailAgentModalComponent {...props} />
            </UserContext.Provider>
          )
        );

        GoogleRecaptchaServiceModule.GoogleRecaptchaService().reset = (): Promise<void> => Promise.resolve();

        expect(screen.getByText('something-wrong-try-again')).toBeInTheDocument();
        expect(screen.queryByTestId('loader-template')).not.toBeInTheDocument();
      });

      it('should display error messages when the fields empty', () => {
        userEvent.click(screen.getByRole('button', { name: 'agent-modal/cta-send-message' }));

        const errorTexts = screen.getAllByText(/^validation(?:_?[a-z]+)*/);

        expect(errorTexts).toHaveLength(3);
      });

      it('should display required error message for name', () => {
        userEvent.type(screen.getByLabelText('email'), 'email@example.com');
        userEvent.type(screen.getByPlaceholderText('phone'), '123456');
        userEvent.click(screen.getByRole('button', { name: 'agent-modal/cta-send-message' }));

        expect(screen.getByLabelText('name')).toHaveErrorMessage('validation/empty-name');
      });

      it('should display required error message for email', () => {
        userEvent.type(screen.getByLabelText('name'), 'Name');
        userEvent.type(screen.getByPlaceholderText('phone'), '123456');
        userEvent.click(screen.getByRole('button', { name: 'agent-modal/cta-send-message' }));

        expect(screen.getByLabelText('email')).toHaveErrorMessage('validation/empty-email');
      });

      it('should display required error message for phone', () => {
        userEvent.type(screen.getByLabelText('name'), 'Name');
        userEvent.type(screen.getByLabelText('email'), 'email@example.com');
        userEvent.click(screen.getByRole('button', { name: 'agent-modal/cta-send-message' }));

        expect(screen.getByPlaceholderText('phone')).toHaveErrorMessage('validation/empty-phone-number');
      });

      it('should display required error message for message', () => {
        userEvent.type(screen.getByLabelText('name'), 'Name');
        userEvent.type(screen.getByLabelText('email'), 'email@example.com');
        userEvent.type(screen.getByPlaceholderText('phone'), '123456');
        userEvent.clear(screen.getByLabelText('message'));
        userEvent.click(screen.getByRole('button', { name: 'agent-modal/cta-send-message' }));

        expect(screen.getByLabelText('message')).toHaveErrorMessage('validation/empty-message');
      });

      it('should display invalid email error message', () => {
        userEvent.type(screen.getByLabelText('name'), 'Name');
        userEvent.type(screen.getByLabelText('email'), 'email');
        userEvent.type(screen.getByPlaceholderText('phone'), '123456');
        userEvent.click(screen.getByRole('button', { name: 'agent-modal/cta-send-message' }));

        expect(screen.getByLabelText('email')).toHaveErrorMessage('validation/not-valid-email');
      });

      it('should display invalid phone number error message', () => {
        userEvent.type(screen.getByLabelText('name'), 'Name');
        userEvent.type(screen.getByLabelText('email'), 'email@example.com');
        userEvent.type(screen.getByPlaceholderText('phone'), 'a123');
        userEvent.click(screen.getByRole('button', { name: 'agent-modal/cta-send-message' }));

        expect(screen.getByPlaceholderText('phone')).toHaveErrorMessage('validation/not-valid-phone-number');
      });
    });
  });
});
