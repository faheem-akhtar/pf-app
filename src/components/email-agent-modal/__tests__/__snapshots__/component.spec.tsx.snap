// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailAgentModalComponent when the user is not exist form submission should form successfully submitted 1`] = `
<div
  class="container"
  data-testid="email-agent-modal-content"
>
  <header
    class="header"
  >
    <h1
      class="title"
    >
      agent-modal/email-form-title
    </h1>
    <div
      class="closeButton"
    >
      <svg
        class="closeIcon"
        viewBox="3 3 10 10"
      >
        <path
          clip-rule="evenodd"
          d="M4.34408 3.17475C4.01717 2.92057 3.54448 2.94367 3.24408 3.24408C2.91864 3.56951 2.91864 4.09715 3.24408 4.42259L6.82149 8L3.24408 11.5774C2.91864 11.9028 2.91864 12.4305 3.24408 12.7559C3.54448 13.0563 4.01717 13.0794 4.34408 12.8252L4.42259 12.7559L8 9.17851L11.5774 12.7559L11.6559 12.8252C11.9828 13.0794 12.4555 13.0563 12.7559 12.7559C13.0814 12.4305 13.0814 11.9028 12.7559 11.5774L9.17851 8L12.7559 4.42259C13.0814 4.09715 13.0814 3.56951 12.7559 3.24408C12.4555 2.94367 11.9828 2.92057 11.6559 3.17475L11.5774 3.24408L8 6.82149L4.42259 3.24408L4.34408 3.17475Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </div>
  </header>
  
  <div
    class="success"
  >
    <svg
      class="successIcon"
      viewBox="0 0 32 32"
    >
      <path
        d="M16,0C7.2,0,0,7.2,0,16c0,8.8,7.2,16,16,16s16-7.2,16-16C32,7.2,24.8,0,16,0z M16,30.7C7.9,30.7,1.3,24.1,1.3,16S7.9,1.3,16,1.3c8.1,0,14.7,6.6,14.7,14.7C30.7,24.1,24.1,30.7,16,30.7z"
        fill="currentColor"
      />
      <polygon
        fill="currentColor"
        points="13.5,20.5 7.8,14.7 6.9,15.7 13.5,22.3 25.1,10.6 24.2,9.7"
      />
    </svg>
    <div
      class="successContent"
    >
      <p>
        thank-you
      </p>
      <p>
        your-message-is-sent
      </p>
    </div>
    <section
      class="container"
    >
      <p
        class="title"
      >
        agent-modal/sign-in-title
      </p>
      <ul
        class="list"
      >
        <li
          class="item"
        >
          <svg
            class="icon"
            viewBox="0 0 18 16"
          >
            <path
              clip-rule="evenodd"
              d="M1.428 8.899l6.232 6.526a1.85 1.85 0 002.688-.008l6.224-6.518c1.931-2.022 1.931-5.488-.168-7.492-2.1-2.004-4.949-1.796-6.825.169L9 2.183l-.58-.607C6.546-.39 3.697-.551 1.597 1.407c-2.1 1.959-2.1 5.47-.168 7.492z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          <p>
            agent-modal/sign-in/saved-properties-label
          </p>
        </li>
        <li
          class="item"
        >
          <svg
            class="icon"
            viewBox="0 0 16 16"
          >
            <path
              d="M16 5.841v8.736c0 .395-.138.731-.415 1.008a1.373 1.373 0 0 1-1.008.416H1.425c-.384 0-.72-.138-1.008-.416A1.296 1.296 0 0 1 0 14.577V5.841c0-.085.029-.152.088-.2.058-.048.178-.157.36-.328.18-.17.304-.277.368-.32.064-.042.202-.154.416-.336.213-.181.42-.346.624-.496l.848-.64c.362-.277.789-.597 1.28-.96l1.728-1.264c.02-.021.176-.138.464-.352.288-.213.5-.368.64-.464.138-.096.33-.202.576-.32.245-.117.448-.17.608-.16.16.011.368.064.624.16s.448.203.576.32c.128.118.336.272.624.464s.442.31.464.352l1.728 1.264c.49.363.917.683 1.28.96.362.278.65.491.864.64.213.15.42.315.624.496.202.182.336.294.4.336.064.043.186.15.368.32.18.171.3.28.36.328a.244.244 0 0 1 .088.2zm-5.04 5.216a414.344 414.344 0 0 0 3.089-2.24.3.3 0 0 0 .112-.192.257.257 0 0 0-.064-.208l-.336-.464a.256.256 0 0 0-.192-.112.237.237 0 0 0-.208.064 441.619 441.619 0 0 1-3.072 2.224c-.022.022-.176.139-.464.352l-.624.464a3.27 3.27 0 0 1-.576.32c-.256.118-.464.171-.624.16a2.145 2.145 0 0 1-.608-.16 2.05 2.05 0 0 1-.576-.32 7.475 7.475 0 0 0-.64-.464c-.288-.192-.443-.309-.464-.352C4.603 9.34 3.579 8.6 2.64 7.905a.257.257 0 0 0-.208-.064.3.3 0 0 0-.192.112l-.336.464a.257.257 0 0 0-.064.208.3.3 0 0 0 .112.192c.49.363 1.52 1.11 3.088 2.24.053.054.202.171.448.352.245.182.437.32.576.416.138.096.33.214.576.352.245.139.474.24.688.304.213.064.437.096.672.096.234 0 .458-.032.672-.096.213-.064.448-.17.704-.32.256-.149.448-.266.576-.352.128-.085.325-.224.592-.416.266-.192.405-.304.416-.336z"
              fill="currentColor"
            />
          </svg>
          <p>
            agent-modal/sign-in/agent-communication-label
          </p>
        </li>
        <li
          class="item"
        >
          <svg
            class="icon"
            viewBox="0 0 12 16"
          >
            <path
              d="M5.714 2.179c1.572 0 2.917.571 4.036 1.714C10.87 5.036 11.429 6.405 11.429 8c0 1.143-.298 2.179-.893 3.107L9.5 10.036c.333-.62.5-1.298.5-2.036 0-1.19-.423-2.214-1.268-3.071-.845-.858-1.851-1.286-3.018-1.286V5.82L2.857 2.893 5.714 0v2.179zm0 10.178V10.18l2.857 2.928L5.714 16v-2.179c-1.571 0-2.916-.571-4.035-1.714C.559 10.964 0 9.595 0 8c0-1.19.298-2.226.893-3.107l1.036 1.071c-.334.62-.5 1.298-.5 2.036 0 1.19.422 2.214 1.267 3.071.846.858 1.852 1.286 3.018 1.286z"
              fill="currentColor"
            />
          </svg>
          <p>
            agent-modal/sign-in/sync-all-devices-label
          </p>
        </li>
      </ul>
      <button
        class="button primary regular fullWidth"
      >
        sign-in
      </button>
      <button
        class="notNowButton"
      >
        not-now
      </button>
    </section>
  </div>
</div>
`;

exports[`EmailAgentModalComponent when the user is not exist should render without throwing any errors 1`] = `
<div
  class="container"
  data-testid="email-agent-modal-content"
>
  <header
    class="header"
  >
    <h1
      class="title"
    >
      agent-modal/email-form-title
    </h1>
    <div
      class="closeButton"
    >
      <svg
        class="closeIcon"
        viewBox="3 3 10 10"
      >
        <path
          clip-rule="evenodd"
          d="M4.34408 3.17475C4.01717 2.92057 3.54448 2.94367 3.24408 3.24408C2.91864 3.56951 2.91864 4.09715 3.24408 4.42259L6.82149 8L3.24408 11.5774C2.91864 11.9028 2.91864 12.4305 3.24408 12.7559C3.54448 13.0563 4.01717 13.0794 4.34408 12.8252L4.42259 12.7559L8 9.17851L11.5774 12.7559L11.6559 12.8252C11.9828 13.0794 12.4555 13.0563 12.7559 12.7559C13.0814 12.4305 13.0814 11.9028 12.7559 11.5774L9.17851 8L12.7559 4.42259C13.0814 4.09715 13.0814 3.56951 12.7559 3.24408C12.4555 2.94367 11.9828 2.92057 11.6559 3.17475L11.5774 3.24408L8 6.82149L4.42259 3.24408L4.34408 3.17475Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </div>
  </header>
  
  <h2
    class="name"
  >
    Title from agent
  </h2>
  <form
    class="form"
    data-testid="email-agent-form"
  >
    <div
      class="fieldContainer"
    >
      <div
        class="container"
      >
        <div
          class="content"
        >
          <input
            aria-invalid="false"
            aria-label="name"
            class="input"
            id="name"
            name="name"
            type="text"
            value=""
          />
          <div
            class="placeholder"
          >
            <label
              for="name"
            >
              name
            </label>
          </div>
        </div>
      </div>
    </div>
    <div
      class="fieldContainer"
    >
      <div
        class="container"
      >
        <div
          class="content"
        >
          <input
            aria-invalid="false"
            aria-label="email"
            class="input"
            id="email"
            name="email"
            type="email"
            value=""
          />
          <div
            class="placeholder"
          >
            <label
              for="email"
            >
              email
            </label>
          </div>
        </div>
      </div>
    </div>
    <div
      class="fieldContainer"
    >
      <div
        class="container"
      >
        <div
          class="container"
          role="button"
          tabindex="0"
        >
          <div
            class="body"
          >
            <div
              class="content"
            >
              <div
                class="countryFlag countryFlag--ae"
              />
              <span>
                +971
              </span>
            </div>
            <svg
              class="icon"
              viewBox="4.5 8 16 8"
            >
              <path
                clip-rule="evenodd"
                d="M4.64645 8.14645C4.47288 8.32001 4.4536 8.58944 4.58859 8.78431L4.64645 8.85355L11.6464 15.8536C11.82 16.0271 12.0894 16.0464 12.2843 15.9114L12.3536 15.8536L19.3536 8.85355C19.5488 8.65829 19.5488 8.34171 19.3536 8.14645C19.18 7.97288 18.9106 7.9536 18.7157 8.08859L18.6464 8.14645L12 14.793L5.35355 8.14645C5.17999 7.97288 4.91056 7.9536 4.71569 8.08859L4.64645 8.14645Z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </div>
          <ul
            class="list"
            role="list"
          >
            <li
              class="item"
              role="listitem"
            >
              <div
                class="countryFlag countryFlag--ae"
              />
              United Arab Emirates
               
              <span
                class="phoneCode"
              >
                (
                +971
                )
              </span>
            </li>
            <li
              class="item"
              role="listitem"
            >
              <div
                class="countryFlag countryFlag--tr"
              />
              Turkey
               
              <span
                class="phoneCode"
              >
                (
                +90
                )
              </span>
            </li>
          </ul>
        </div>
        <div
          class="content"
        >
          <input
            aria-invalid="false"
            aria-label="phone"
            class="input"
            id="phone-number"
            maxlength="12"
            name="phone"
            placeholder="phone"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      class="fieldContainer"
    >
      <div
        class="container container--textarea"
      >
        <div
          class="content"
        >
          <textarea
            aria-invalid="false"
            aria-label="message"
            class="input input--floating input--textarea"
            id="message"
            name="message"
            type="text"
          >
            Hi, I found your property with ref: 123 on Property Finder. Please contact me. Thank you.
          </textarea>
          <div
            class="placeholder placeholder--floating"
          >
            <label
              for="message"
            >
              message
            </label>
          </div>
        </div>
      </div>
    </div>
    <div
      id="recaptcha"
    />
    <div
      class="fieldContainer"
    >
      <div
        class="container"
      >
        <input
          aria-labelledby="emailAlert"
          checked=""
          class="input"
          id="email-alert"
          name="emailAlert"
          type="checkbox"
        />
        <label
          class="label"
          for="email-alert"
        >
          agent-modal/email-alert-message
        </label>
      </div>
    </div>
    <button
      class="button primary regular button"
    >
      agent-modal/cta-send-message
    </button>
  </form>
</div>
`;
